File: application/src/main/java/ru/melodin/fast/api/method/AppMethodSetter.java
package ru.melodin.fast.api.method;

/**
 * Method setter for apps
 */

import ru.melodin.fast.util.ArrayUtil;

public class AppMethodSetter extends MethodSetter {

    public AppMethodSetter(String name) {
        super(name);
    }

    public AppMethodSetter appId(int id) {
        put("app_id", id);
        return this;
    }

    public AppMethodSetter appIds(int... ids) {
        put("app_ids", ArrayUtil.toString(ids));
        return this;
    }

    public AppMethodSetter extended(boolean value) {
        put("extended", value);
        return this;
    }

    public AppMethodSetter returnFriends(boolean friends) {
        put("return_friends", friends);
        return this;
    }

}

File: app/src/main/java/ru/melodin/fast/api/method/AppMethodSetter.kt
package ru.melodin.fast.api.method

import ru.melodin.fast.util.ArrayUtil

class AppMethodSetter(name: String) : MethodSetter(name) {

    fun appId(id: Int): AppMethodSetter {
        put("app_id", id)
        return this
    }

    fun appIds(vararg ids: Int): AppMethodSetter {
        put("app_ids", ArrayUtil.toString(*ids))
        return this
    }

    override fun extended(value: Boolean): AppMethodSetter {
        put("extended", value)
        return this
    }

    fun returnFriends(friends: Boolean): AppMethodSetter {
        put("return_friends", friends)
        return this
    }

}